### PR TITLE
RDK-45023: Dual ODM support for Pioneer 2K

### PR DESCRIPTION
**Pull request by lakshmiravishankar**

### CODE REVIEW

File: RDKShell/CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,11 @@ if (PLUGIN_RDKSHELL_IGNORE_PS_FLAG_ON_MBFTA)
   add_definitions("-DRDKSHELL_IGNORE_PS_FLAG_ON_MBFTA")
 endif (PLUGIN_RDKSHELL_IGNORE_PS_FLAG_ON_MBFTA)
 
+if (PLUGIN_RDKSHELL_DUAL_ODM_SUPPORT)
+  add_definitions("-DRDKSHELL_DUAL_ODM_SUPPORT")
+  add_definitions("-DDUAL_ODM")
```
Comment (madanagopalt) [inline RDKShell/CMakeLists.txt:64]: what is need for define DUAL_ODM?. please clarify

File: RDKShell/RDKShell.cpp [hunk 2]
```diff
@@ -271,10 +275,23 @@ static bool checkAssemblyFactoryMode_wrapper()
         bool ret = false;
         Device_Mode_Result_t result = Device_Mode_getFactoryMode(&factoryMode);
         if(result == DEVICE_MODE_RESULT_SUCCESS) {
+#ifdef RDKSHELL_DUAL_ODM_SUPPORT
```
Comment (madanagopalt) [inline RDKShell/RDKShell.cpp:278]: please help to move complete checks to a different function of getting factory app url.

File: RDKShell/RDKShell.cpp [hunk 3]
```diff
@@ -5329,7 +5346,39 @@ namespace WPEFramework {
...
+                    {
+                        factoryAppUrl = getenv("RDKSHELL_ASSEMBLY_FACTORY_APP_URL");
+                    }
+                    else if ((sFactoryMode == DEVICE_MODE_ODM_2_FACTORY_MODE) && (isOdmEnabled == true))
+                    {
+                        factoryAppUrl = getenv("RDKSHELL_ASSEMBLY_FACTORY_2_APP_URL");
+                    }
+		    else if (isOdmEnabled == false)
+                    {
+                           Device_Mode_FactoryModes_t factoryMode;
```
Comment (madanagopalt) [inline RDKShell/RDKShell.cpp:5361]: what will be default url, if any of below condition fails?

**Review by madanagopalt — APPROVED** [review]
looks good to me

**Review by apatel859 — APPROVED** [review]

**Pull request merged**